plugins/api/include/irods/plugins/api/switch_user_types.h

```diff
@@ -1,15 +1,63 @@
 #ifndef IRODS_SWITCH_USER_TYPES_H
 #define IRODS_SWITCH_USER_TYPES_H
 
+/// The input data type used to invoke rc_switch_user.
+///
+/// \since 4.3.1
 typedef struct SwitchUserInput // NOLINT(modernize-use-using)
 {
-    // NOLINTBEGIN(modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
+    ///
+    /// \since 4.3.1
     char username[64];
+
+    /// The part of a fully-qualified iRODS username following the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    ///
+    /// \since 4.3.1
     char zone[64];
-    // NOLINTEND(modernize-avoid-c-arrays)
+
+    // NOLINTEND(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    // NOLINTEND(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
+    /// Influences how the server handles the proxy user.
+    ///
+    /// If set to 1, the proxy user is updated. This option will result in losing the
+    /// ability to invoke rc_switch_user if this option is set to 1 and the user to
+    /// switch to is not a \p rodsadmin. This option is not used when handling
+    /// server-to-server connections.
+    ///
+    /// \since 4.3.1
+    int update_proxy_user;
```

Comment:
I feel like a word other than "update" may be better here... Does `switch_proxy_user` seem okay? If "update" is clear to everyone, I think it's fine. Just felt a little vague to me.

Reply:
I'm fine with either, but I prefer the one that is the clearest.

Given the name of the API uses `switch`, it seems more correct to use `switch` here as well.

Are there other words/phrases that are better?

Reply:
Changed to `switch_proxy_user`.